src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1724,4 +1733,256 @@ void scaleInletFlows(EnergyPlusData &state, int const zoneNodeNum, int const spa
     spaceNode.MassFlowRateMin = zoneNode.MassFlowRateMin * frac;
     spaceNode.MassFlowRateMinAvail = zoneNode.MassFlowRateMinAvail * frac;
 }
+
+void ZoneEquipmentSplitterMixer::size(EnergyPlusData &state)
+{
+    bool anyAutoSize = false;
+    for (auto &thisSpace : this->spaces) {
+        if (thisSpace.fraction == DataSizing::AutoSize) {
+            anyAutoSize = true;
+            break;
+        }
+    }
+    if (!anyAutoSize) return;
+
+    // Calculate total of space fraction basis value across all spaces for this splitter or mixer
+    // including spaces which are not autosized here.
+    Real64 spacesTotal = 0.0;
+    switch (this->spaceSizingBasis) {
+    case DataZoneEquipment::SpaceEquipSizingBasis::DesignCoolingLoad:
+        for (auto &thisSpace : this->spaces) {
+            spacesTotal += state.dataSize->FinalSpaceSizing(thisSpace.spaceIndex).DesCoolLoad;
+        }
+        break;
+    case DataZoneEquipment::SpaceEquipSizingBasis::DesignHeatingLoad:
+        for (auto &thisSpace : this->spaces) {
+            spacesTotal += state.dataSize->FinalSpaceSizing(thisSpace.spaceIndex).DesHeatLoad;
+        }
+        break;
+    case DataZoneEquipment::SpaceEquipSizingBasis::FloorArea:
+        for (auto &thisSpace : this->spaces) {
+            spacesTotal += state.dataHeatBal->space(thisSpace.spaceIndex).FloorArea;
+        }
+        break;
+    case DataZoneEquipment::SpaceEquipSizingBasis::Volume:
+        for (auto &thisSpace : this->spaces) {
+            spacesTotal += state.dataHeatBal->space(thisSpace.spaceIndex).Volume;
+        }
+        break;
+    case DataZoneEquipment::SpaceEquipSizingBasis::PerimeterLength:
+        ShowFatalError(state,
+                       format("ZoneEquipmentSplitterMixer::size: Space Fraction Method={} not supported for {}={}",
+                              DataZoneEquipment::spaceEquipSizingBasisNamesUC[(int)this->spaceSizingBasis],
+                              BranchNodeConnections::ConnectionObjectTypeNames[(int)this->spaceEquipType],
+                              this->Name));
+        break;
+    default:
+        break;
+    }
+
+    if (spacesTotal == 0.0) {
```

Comment:
Does anything break down if it's 0.000001?  Or is it fine?

Reply:
> Does anything break down if it's 0.000001? Or is it fine?

Probably. `spacesTotal` gets used as a divisor. I'll make something small just in case.